evm.md

```diff
@@ -182,7 +185,78 @@ Our semantics is modal, with the initial mode being set on the command line via
 
 State Stacks
 ------------
+### The BlockchainStack
+
+The `blockList` cell stores a list of previous block and network states.
+-   `#pushBlockchainState` saves a copy of the block state and network state on the `blockList`.
+-   `#getBlockchainState(Int)` restores a blockchain state for a given block number.
+
+```k
+    syntax BlockchainItem ::= "{" NetworkCellFragment "|" BlockCell "}"
+ // -------------------------------------------------------------------
 
+   syntax InternalOp ::= "#pushBlockchainState"
+ // ---------------------------------------
+    rule <k> #pushBlockchainState => . ... </k>
+         <blockList> (.List => ListItem({ NETWORK | <block> BLOCK </block> })) ... </blockList>
+         <network> NETWORK </network>
+         <block>   BLOCK   </block>
+
+   syntax InternalOp ::= "#getBlockchainState" "(" Int ")"
+ // ---------------------------------------
+    rule <k> #getBlockchainState(BLOCKNUM) => . ... </k>
+         <blockList> (
+            ListItem(
+               { NETWORK |
+             (
+                <block>
+                  <previousHash>     HP       </previousHash>     // I_Hp
+                  <ommersHash>       HO       </ommersHash>       // I_Ho
+                  <coinbase>         HC       </coinbase>         // I_Hc
+                  <stateRoot>        HR       </stateRoot>        // I_Hr
+                  <transactionsRoot> HT       </transactionsRoot> // I_Ht
+                  <receiptsRoot>     HE       </receiptsRoot>     // I_He
+                  <logsBloom>        HB       </logsBloom>        // I_Hb
+                  <difficulty>       HD       </difficulty>       // I_Hd
+                  <number>           BLOCKNUM </number>           // I_Hi
+                  <gasLimit>         HL       </gasLimit>         // I_Hl
+                  <gasUsed>          HG       </gasUsed>          // I_Hg
+                  <timestamp>        HS       </timestamp>        // I_Hs
+                  <extraData>        HX       </extraData>        // I_Hx
+                  <mixHash>          HM       </mixHash>          // I_Hm
+                  <blockNonce>       HN       </blockNonce>       // I_Hn
+
+                  <ommerBlockHeaders> OBH </ommerBlockHeaders>
+                  <blockhash>         BH  </blockhash>
+                </block>
```

Comment:
We definitely sohuld not have to mention all this state. We should be able to say `<block> BLOCK </block>`

Reply:
Or, if you use `BlockCellFragment` instead, just `{ NETWORK | BLOCK }`